### DISCUSSION TITLE
Issues Changing Background Image

### DISCUSSION THREAD

**Discussion by Tim-K5AC** [aqualinkd/AqualinkD#339] [General]
I have the following set in /var/www/aqualinkd/config.js

var background_url='hk/POOL.jpg
var background_reload = 0; 

My .jpg is located in the same folder as the default background.jpg, but I can't seem to get it to load.  Are there any limitations on the background?  My .jpg is only a few hundred Kbytes.  I've rebooted, refreshed, etc. and no luck.   Any insights?

Thanks for your help!

Tim P

**Comment by sfeakes** [comment]
Look at your line again.
`var background_url='hk/POOL.jpg`
should be
`var background_url='hk/POOL.jpg';`

you also don't need `var background_reload = 0;` when using a static image. (but won't cause an issue)

Assuming the above was a cut n paste issue.    If loading the background fails, it will go back to default `background.jpg`

What happens when you use the below URL directly?   
http://<aqualinkd.ip.address>/hk/POOL.jpg

Do you see the file using following `ls -al /var/www/aqualinkd/hk/POOL.jpg`

**Comment by Tim-K5AC** [comment]
I was pressing the refresh button in the Chromium browser, this did nothing.

when I pressed a Ctrl-F5 I got my background image!  However, my spa light and pool light disappeared???

**Comment by sfeakes** [comment]
I would check the syntax in your `config.js` file.  Or go back to original file and make sure all is working and start again.

**Reply by Tim-K5AC** [reply]
I did a clean and install latest and I still do not have my lights.  I see a message on the logger under iAQTouch:  Don't know where to place button 1.

I have an RS8 (version t.2) with an All Touch RS-4 (that I don't use, but works correctly).     Neither one of the lights show up in the Web Browser now, but I noticed that the button 1/pump now shows wattage, which it didn't do with the prior attempt.

**Reply by sfeakes** [reply]
Can you post the following files
/etc/aqualinkd.conf
/var/www/aqualinkd/config.js

and also your log file) after starting and leaving running for a minute).  Use the download option in aqmanager.

**Reply by Tim-K5AC** [reply]
The files are located here:
https://drive.google.com/drive/folders/1rwOHJGT12qe6AdjikyG2COiBU8G2imrM?usp=drive_link

**Reply by sfeakes** [reply]
I don’t have access.  You can upload them to this chat with the paperclip.

**Comment by Tim-K5AC** [comment]
[aqualinkd.log](https://github.com/user-attachments/files/16875359/aqualinkd.log)

**Comment by Tim-K5AC** [comment]
[aqualinkd.txt](https://github.com/user-attachments/files/16875454/aqualinkd.txt)
[config.txt](https://github.com/user-attachments/files/16875455/config.txt)

**Reply by Tim-K5AC** [reply]
It wouldn't let me paperclip them, said file extension not supported.  I renamed to .txt

**Comment by sfeakes** [comment]
Quick Look at config.js and you have told it to hide just about all buttons except filter pump and spa.
You hide by using the button ID not the Label.   
uncomment the aux_4 to aux_7.

I’ll look at the other files later when I have more time.

**Reply by Tim-K5AC** [reply]
I put Aux_4 to Aux_7 back and did a hard re-fresh, but still no SPA nor POOL lights in the browser.

**Comment by sfeakes** [comment]
Changing config.js does not require a system restart or aqualinkd restart. BUT does require a browser refresh of cache, the browser will cache config.js until it’s told to reset.  Clear browser cache or force a reload of that file.  On some browsers simply going to the config.js file directly will do that.  Use the below url and hit refresh until you get the latest version, then check again.  
`http://<aqualinkd.ip.address>/config.js`

**Reply by Tim-K5AC** [reply]
I did that and still no lights.  What is the error/message in the logger that says iAqTouch don't know what to do with button 1?

**Reply by sfeakes** [reply]
That error can be ignored, it should be a notification not error.  It’s simply pulling the information from the AqualinkTouch protocol and it is seeing a button that I have not coded around.  The buttons on that protocol have nothing to do with the buttons in the AqualinkD web UI.

**Reply by sfeakes** [reply]
Just looked at your config.js again,   You deleted Aux_2 and Aux_3 from the file.   
You comment them out not delete or rename the lines.

You are probably best off re-getting `config.js` and starting again.

**Reply by Tim-K5AC** [reply]
Yes, THAT was the issue.   For some reason, I replaced the Aux-# with the name of the function (SPA Light).  Now resolved.  Thanks!
